server/sonar-server/src/main/java/org/sonar/server/component/ws/SuggestionsAction.java

```diff
@@ -200,6 +202,12 @@ private SuggestionsWsResponse loadSuggestionsWithoutSearch(int skip, int limit,
   }
 
   private SuggestionsWsResponse loadSuggestionsWithSearch(String query, int skip, int limit, Set<String> recentlyBrowsedKeys, List<String> qualifiers) {
+    if (split(query).noneMatch(token -> token.length() >= MINIMUM_NGRAM_LENGTH)) {
+      SuggestionsWsResponse.Builder queryBuilder = newBuilder();
+      getWarning(query).ifPresent(queryBuilder::setWarning);
```

Comment:
Having a single call to `getWarning()` in this method would be better but ok, it's acceptable.